### DISCUSSION TITLE
How to get the parameters of `keras_model_sequential` model?

### DISCUSSION THREAD

**Discussion by danli349** [rstudio/keras#1483] [General]
Hello: 

How to get the parameters of `keras_model_sequential` model? 
Thanks 

```{r}
model <- keras_model_sequential() %>%
  layer_embedding(input_dim = 10000, output_dim = 32) %>%
  layer_lstm(units = 32) %>%
  layer_dense(units = 1, activation = "sigmoid")
summary(model)
```

![image](https://github.com/user-attachments/assets/40fa0d11-857e-41a2-8a71-dc1cbee6c646)

```{r}
get_layer(model, "lstm_1")$weights
```
empty

**Comment by t-kalinowski** [comment]
You can specify a sequential model's input shape and dtype by passing arguments to `keras_model_sequential()`. See https://keras3.posit.co/reference/keras_model_sequential.html

**Reply by danli349** [reply]
@t-kalinowski I see, Thanks a lot

**Reply by danli349** [reply]
@t-kalinowski 

I am sorry, Could you let me know why the `kernel` and `recurrent_kernel` shape is ([32, 128])? 
The `layer_embedding` output shape is ([32,]), I think the `kernel` and `recurrent_kernel` shape would be ([32, 32]). 
Thanks 

```{r}
model <- keras_model_sequential(input_shape = c(500)) %>%
  layer_embedding(input_dim = 10000, output_dim = 32) %>%
  layer_lstm(units = 32) %>%
  layer_dense(units = 1, activation = "sigmoid")
summary(model)
```
![image](https://github.com/user-attachments/assets/d6320451-50c5-4f46-9391-35f1930f0114)

```{r}
get_layer(model, "lstm")$weights |> sapply(\(w) setNames(list(w$shape), w$name)) 

$kernel
TensorShape([32, 128])

$recurrent_kernel
TensorShape([32, 128])

$bias
TensorShape([128])
```

**Reply by t-kalinowski** [reply]
It might help to work though what work the LSTM layer is performing. In the "Deep Learning with R" book there is a "lstm from scratch" section with example code: https://github.com/t-kalinowski/deep-learning-with-R-2nd-edition-code/blob/5d666f93d52446511a8a8e4eb739eba1c0ffd199/ch10.R#L327-L342

**Reply by danli349** [reply]
@t-kalinowski I will take a look. Thanks
